fuel-relayer/src/relayer.rs

```diff
@@ -278,95 +213,110 @@ impl Relayer {
             // If not the same, stop listening to events and do 2,3,4 steps again.
             // empty pending and do overlaping sync again.
             // Assume this will not happen very often.
-            self.pending.clear();
         }
 
         // 5. Continue to active listen on eth events. and prune(commit to db) dequeue for older finalized events
-        let finalized_da_height = best_block.as_u64() - self.config.eth_finality_period();
-        self.pending
+        let finalized_da_height =
+            best_block.as_u64() as DaBlockHeight - self.config.da_finalization();
+        self.queue
             .commit_diffs(self.db.as_mut(), finalized_da_height)
             .await;
 
-        watchers.ok_or_else(|| RelayerError::ProviderError.into())
+        watchers
+            .map(|(w1, w2)| (best_block.as_u64() as DaBlockHeight, w1, w2))
+            .ok_or_else(|| RelayerError::ProviderError.into())
     }
 
     /// Starting point of relayer
-    #[tracing::instrument(name = "relayer_run", skip_all)]
-    pub async fn run<P>(mut self, provider: P)
+    #[tracing::instrument(name = "main", skip_all)]
+    pub async fn run<P>(mut self, provider: Arc<P>)
     where
-        P: Middleware<Error = ProviderError>,
+        P: Middleware<Error = ProviderError> + 'static,
     {
-        //let mut this = self;
-        self.current_validator_set
-            .load_get_validators(self.db.as_mut())
-            .await;
+        self.queue.load_validators(self.db.as_mut()).await;
 
-        loop {
-            // initial sync
-            let (mut blocks_watcher, mut logs_watcher) = match self.initial_sync(&provider).await {
-                Ok(watcher) => watcher,
+        let mut number_of_tries = 10;
```

Comment:
10 is a magic number. Max retries should always be configurable constants.

Reply:
introduced constant in config